source/Background/Controller.js

```diff
@@ -380,30 +381,86 @@ backgroundController.exposeController(
 );
 
 backgroundController.exposeController(
-  'sign',
-  async (opts, payload, metadata) => {
-    const { callback } = opts;
+  'requestSign',
+  async (opts, payload, metadata, requestInfo) => {
+    const { message, sender, callback } = opts;
+    const { id: callId } = message.data.data;
+    const { id: portId } = sender;
+    const { canisterId, requestType } = requestInfo;
     try {
       storage.get(keyring.currentWalletId.toString(), async (state) => {
-        const apps = state?.[keyring.currentWalletId]?.apps || {};
+        const app = state?.[keyring.currentWalletId]?.apps?.[metadata.url] || {};
         if (
-          apps?.[metadata.url]?.status !== CONNECTION_STATUS.accepted
+          app.status !== CONNECTION_STATUS.accepted
         ) {
           callback(ERRORS.CONNECTION_ERROR, null);
           return;
         }
-        const parsedPayload = payload instanceof Buffer
-          ? payload
-          : Buffer.from(Object.values(payload));
-        const signed = await keyring.sign(parsedPayload);
-        callback(null, [...new Uint8Array(signed)]);
+
+        if (requestType !== 'read_state' && canisterId && !(canisterId in Object.values(app.whitelist))) {
```

Comment:
what's the whitelst format? thought it was an array

Reply:
with what format?

Reply:
```
{
  [canisterId]: { canisterDabInfo }
}
```